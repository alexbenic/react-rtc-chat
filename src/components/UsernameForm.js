import React, { Component } from "react";

class UsernameForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: ""
    };
    this.onSubmit = this.onSubmit.bind(this);
    this.onChange = this.onChange.bind(this);
  }

  onSubmit(e) {
    e.preventDefault();
    this.props.onSubmit(this.state.username);
  }

  onChange(e) {
    this.setState({ username: e.target.value });
  }

  render() {
    return (
      <div className="vh-100 vw-100 flex flex-column items-center">
        <form className="flex flex-column f-form" onSubmit={this.onSubmit}>
          <span className="f2 b tc c-react"> Choose nick </span>
          <input
            tabIndex="0"
            className="mt3 mb3 outline-0 f-input br3 tc f3 pa2"
            type="text"
            onChange={this.onChange}
          />
          <button
            className="f3 w-30 center outline-0 bg-react white bn br3 pa2"
            type="submit"
          >
            Enter
          </button>
        </form>
      </div>
    );
  }
}

export default UsernameForm;
