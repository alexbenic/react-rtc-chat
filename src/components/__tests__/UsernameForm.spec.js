import React from "react";
import each from "jest-each";
import { render, cleanup, fireEvent } from "react-testing-library";
import UsernameForm from "../UsernameForm";

afterEach(cleanup);

// prettier-ignore
each([
  "zvrk",
  "zook",
  "maci",
  "mcmc",
])
.test("should submit user: %s", nick => {
  const handleSubmit = jest.fn();
  const { container, getByPlaceholderText } = render(
    <UsernameForm onSubmit={handleSubmit} />
  );

  const input = getByPlaceholderText('nick');
  const btn = container.querySelector('button')
  input.value = nick;

  fireEvent.change(input)
  fireEvent(btn, new Event("click"));

  expect(handleSubmit).toHaveBeenCalledTimes(1);
  expect(handleSubmit).toHaveBeenCalledWith(nick);
});

test("username form snapshot", () => {
  const handleSubmit = jest.fn();
  const { container } = render(<UsernameForm onSubmit={handleSubmit} />);

  expect(container).toMatchSnapshot();
});
