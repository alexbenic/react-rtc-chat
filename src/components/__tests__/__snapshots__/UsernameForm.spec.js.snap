// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`username form snapshot 1`] = `
<div>
  <div
    class="vh-100 vw-100 flex flex-column items-center"
  >
    <form
      class="flex flex-column f-form"
    >
      <span
        class="f2 b tc c-react"
      >
        Choose nick 
      </span>
      <input
        tabindex="0"
        placeholder="nick"
        class="mt3 mb3 outline-0 f-input br3 tc f3 pa2"
        type="text"
      />
      <button
        class="f3 w-30 center outline-0 bg-react white bn br3 pa2"
        type="submit"
      >
        Enter
      </button>
    </form>
  </div>
</div>
`;
